featuretools/tests/primitive_tests/test_distancetoholiday_primitives.py

```diff
@@ -0,0 +1,62 @@
+from datetime import datetime
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from featuretools.primitives.standard.transform_primitive import (
+    DistanceToHoliday
+)
+
+
+def test_distanceholiday():
+    distance_to_holiday = DistanceToHoliday("New Year's Day")
+    dates = [datetime(2010, 1, 1),
+             datetime(2012, 5, 31),
+             datetime(2017, 7, 31),
+             datetime(2020, 12, 31)]
+
+    distance_list = distance_to_holiday(dates).tolist()
+
+    assert distance_list[0] == 0
+    assert distance_list[1] == -151
+    assert distance_list[2] == 154
+    assert distance_list[3] == 1
+
+
+def test_unknown_country_error():
+    error_text = r"must be one of the available countries.*"
+    with pytest.raises(ValueError, match=error_text):
+        DistanceToHoliday("Victoria Day", country='UNK')
+
+
+def test_unknown_holiday_error():
+    error_text = r"must be one of the available holidays.*"
+    with pytest.raises(ValueError, match=error_text):
+        DistanceToHoliday("Alteryx Day")
+
+
+def test_nat():
+    date_to_holiday = DistanceToHoliday("New Year's Day")
+    case = pd.Series([
+        '2010-01-01',
+        'NaT',
+        '2012-05-31',
+        'NaT',
+    ]).astype('datetime64')
+    answer = [0, np.nan, -151, np.nan]
+    given_answer = date_to_holiday(case).astype('float')
+    np.testing.assert_array_equal(given_answer, answer)
+
+
+def test_valid_country():
```

Comment:
I think the original version of this test was perhaps testing something slightly different since it was named `test_holiday_out_of_range`, but I'm not exactly clear now what the intent of that test was. Do you have any ideas? I see the original test introduced `np.nan` values in the result, but this one does not.

Reply:
@thehomebrewnerd I dug into this a bit. The test above isn't replacing that test, I actually didn't see the original tests when moving things over. 

This out of range test brought to my attention some unexpected behavior when using the holidays library. It appears the observed holidays are only present in the dataset. For example, the test date of 2010-1-1 results in a NaN because there is no holiday named "Boxing Day" in the data set for 2009 and 2010 since they fall on Saturday and Sunday respectively. Instead, the dataset contains "Boxing Day (Observed)".

I think this is confusing behavior in my opinion